Modify the splash page

aks681 opened the pull request:
## Overview

1. This PR does the following: 

- Modifies the splash page according to the new mocks: [here](https://www.figma.com/file/V7liQVX89cwbaWFcAaZGGp/Oppia-Splash-page-different-images?node-id=74%3A7).

Desktop + Mobile Video Link: [here](https://drive.google.com/file/d/14-4Y9kIWYQ0-yly4ppQVz8OZbmcv23tv/view?usp=sharing)

## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [ ] The linter/Karma presubmit checks have passed locally on your machine.
- [ ] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [ ] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Make sure to follow the [instructions for making a code change](https://github.com/oppia/oppia/wiki/Contributing-code-to-Oppia#instructions-for-making-a-code-change).
- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays.
- Never force push. If you do, your PR will be closed.

aks681 committed:
4 sections done

aks681 committed:
Merge remote-tracking branch 'upstream/develop' into splash-page

aks681 committed:
minor change

aks681 committed:
Desktop page done

aks681 committed:
lint

aks681 committed:
fixed lint and TS errors

aks681 committed:
minor change

seanlip commented:
Hi @aks681, this looks great, thanks! Though please note that all the CI checks seem to be failing. Also we should complete the mobile view before this gets merged.

@dchen97, could you please review this as well? (Mainly the desktop video link [here](https://drive.google.com/file/d/1lr7QRHmQmcqw2pqaFgrDFhib04sjCb6P/view?usp=sharing))

Also @aks681, some things I noticed:

- Change "100 Curated Lessons" to "40 Curated Lessons", since that's where things are at the moment. We'll increase it as we add more.
- At the end of "An Immersive Environment", change "check" to "Check".
- I'm going to chat with @dchen97 about updating the Testimonials section so that we can get a more diverse representation, so just wanted to let you know that that might change a bit (though this doesn't block merging).

Thanks!

dchen97 commented:
Thanks for putting this together! Overall I think this looks great. I just
have a couple of notes below:

   - Change the headline from "Oppia: an Education Platform for Everyone"
   to "Oppia: Learning for Everyone" - education platform is too technical and
   is a bit more wordy than I think is necessary here
   - Remove the period from the end of "Engaging, effective, personalized."
   - Similarly to the issue I had submitted earlier, "Browse Our Lessons"
   should link to the Math Classroom instead of the Library. If people think
   it would be confusing, we can change this to "Browse our Math Lessons"
   and/or potentially add a separate CTA (that is not a primary CTA) for the
   Library (e.g. "Browse Community Library"), but ultimately the main CTA
   should direct to the Classroom.

+1 to needing more diversity in the quotes, but I'll talk more with Sean
about how we can get those for you.

Thanks again!

*Diana Chen*
*Growth Team Lead | Oppia Foundation*
*E*: diana@oppia.org




On Tue, Nov 24, 2020 at 11:47 AM Sean Lip <notifications@github.com> wrote:

> Hi @aks681 <https://github.com/aks681>, this looks great, thanks! Though
> please note that all the CI checks seem to be failing. Also we should
> complete the mobile view before this gets merged.
>
> @dchen97 <https://github.com/dchen97>, could you please review this as
> well? (Mainly the desktop video link here
> <https://drive.google.com/file/d/1lr7QRHmQmcqw2pqaFgrDFhib04sjCb6P/view?usp=sharing>
> )
>
> Also @aks681 <https://github.com/aks681>, some things I noticed:
>
>    - Change "100 Curated Lessons" to "40 Curated Lessons", since that's
>    where things are at the moment. We'll increase it as we add more.
>    - At the end of "An Immersive Environment", change "check" to "Check".
>    - I'm going to chat with @dchen97 <https://github.com/dchen97> about
>    updating the Testimonials section so that we can get a more diverse
>    representation, so just wanted to let you know that that might change a bit
>    (though this doesn't block merging).
>
> Thanks!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/oppia/oppia/pull/11260#issuecomment-733195370>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC32BWYVZO2PO3B26TT6CATSRQEVZANCNFSM4UBJ2KKQ>
> .
>

aks681 committed:
Merge remote-tracking branch 'upstream/develop' into splash-page

aks681 committed:
added mobile view

aks681 commented:
Updated PR description with video of desktop + mobile UI. Addressed above comments as well.

seanlip commented:
@vojtechjelinek @kevintab95 @nithusha21 @dchen97 @srijanreddy98 @jameesjohn PTAL as codeowners. Thanks!

seanlip commented:
Also, @aks681, please note the failing backend/frontend/lint tests.

seanlip commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -61,16 +64,78 @@ angular.module('oppia').component('splashPage', {
+          communityBackgroundImageName = 'm_community_background.png';
+          testimonialBackgroundImageName = 'm_testimonial_background.png';
+        }
+        Array.from(
```
This seems pretty fragile. Could we not do it this way? Instead, maybe use CSS or ng-class or something similar?

aks681 replied:
Done, used ng-style.

seanlip reviewed (changes requested):
@aks681 Took a pass, I have just one inline comment.

Also ... why are there a bunch of empty ":Zone.Identifier" files?

seanlip commented on assets/i18n/en.json:
```diff
@@ -519,16 +519,41 @@
+    "I18N_SPLASH_TEACHERS_CONTENT": "With Oppia's content creation system, you can easily customize the lesson's format for your students. Create and share lessons on topics you are passionate about.",
+    "I18N_SPLASH_TEACHERS_TITLE": "Easily Share Your Knowledge",
+    "I18N_SPLASH_TESTIMONIAL_1": "\"I enjoyed playing the lesson a lot, I did not feel bored and I feel like I master the negative numbers now\"",
+    "I18N_SPLASH_TESTIMONIAL_2": "\"I learned ratios at school but with Oppia I introduced to many new things like the three items ratios and combining ratios\"",
```
I introduced --> I was introduced

aks681 replied:
Done

seanlip commented on assets/i18n/en.json:
```diff
@@ -519,16 +519,41 @@
+    "I18N_SPLASH_TEACHERS_TITLE": "Easily Share Your Knowledge",
+    "I18N_SPLASH_TESTIMONIAL_1": "\"I enjoyed playing the lesson a lot, I did not feel bored and I feel like I master the negative numbers now\"",
+    "I18N_SPLASH_TESTIMONIAL_2": "\"I learned ratios at school but with Oppia I introduced to many new things like the three items ratios and combining ratios\"",
+    "I18N_SPLASH_TESTIMONIAL_3": "\"I love Oppia so much, and I hope that more topics like geographic  and science are available in Arabic\"",
```
geographic --> geography

Drop extra space before "and".

aks681 replied:
Done

dchen97 commented:
Thanks for putting this together! My only note is that after looking it over, I think we should change the subheading "Effective, Engaging, Personalized" to `Engaging and effective quality education for all` - it provides more context and is a bit less technical. Otherwise it LGTM

vojtechjelinek commented on assets/i18n/en.json:
```diff
@@ -519,16 +519,41 @@
+    "I18N_SPLASH_SUBTITLE": "Engaging, effective, personalized",
+    "I18N_SPLASH_TEACHERS_CONTENT": "With Oppia's content creation system, you can easily customize the lesson's format for your students. Create and share lessons on topics you are passionate about.",
+    "I18N_SPLASH_TEACHERS_TITLE": "Easily Share Your Knowledge",
+    "I18N_SPLASH_TESTIMONIAL_1": "\"I enjoyed playing the lesson a lot, I did not feel bored and I feel like I master the negative numbers now\"",
```
Maybe use proper English typographical quotes `“` (left) `”` (right).

aks681 replied:
Done

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -61,16 +64,78 @@ angular.module('oppia').component('splashPage', {
       };
 
+      ctrl.onClickLearnMoreStudentsButton = function() {
+        $timeout(function() {
+          WindowRef.nativeWindow.location = '/learn/math';
+        }, 150);
```
Why is this timeout needed here? Since the button is only used as a link should we maybe use `<a>` and change the appearance of it as recommended here https://stackoverflow.com/a/2906586/3688189

aks681 replied:
Done, used the method in the link.

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -61,16 +64,78 @@ angular.module('oppia').component('splashPage', {
         return false;
       };
+
+      ctrl.incrementTestimonialId = function() {
+        ctrl.testimonialId = (ctrl.testimonialId + 1) % 3;
+      };
+
+      ctrl.decrementTestimonialId = function() {
+        ctrl.testimonialId = (((ctrl.testimonialId - 1) % 3) + 3) % 3;
+      };
```
Maybe explain the calculations here with a comment.

aks681 replied:
Done

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -61,16 +64,78 @@ angular.module('oppia').component('splashPage', {
         ctrl.userIsLoggedIn = null;
+        ctrl.testimonialId = 0;
         LoaderService.showLoadingScreen('Loading');
+        var communityBackgroundImageName = 'dsk_community_background.png';
+        var testimonialBackgroundImageName = 'dsk_testimonial_background.png';
+        if (ctrl.isWindowNarrow()) {
+          communityBackgroundImageName = 'm_community_background.png';
+          testimonialBackgroundImageName = 'm_testimonial_background.png';
```
Maybe add these to constants.

aks681 replied:
Changed this method to have separate functions that return complete URLs.

vojtechjelinek reviewed (changes requested):
Thanks. Left a few comments.

aks681 committed:
Merge remote-tracking branch 'upstream/develop' into splash-page

aks681 committed:
Addressed review comments

aks681 committed:
fixed backend test

aks681 committed:
fixed frontend test

seanlip commented on assets/i18n/en.json:
```diff
@@ -519,16 +519,41 @@
+    "I18N_SPLASH_FOR_TEACHERS": "For Teachers",
+    "I18N_SPLASH_FOR_VOLUNTEERS": "For Volunteers",
+    "I18N_SPLASH_ICON_ONE_TEXT": "1 Million+ Users",
+    "I18N_SPLASH_ICON_THREE_TEXT": "40 Curated Lessons",
```
This number is going to change ... should we put 40 in a variable instead and have it be interpolated accordingly in the translation, similar to `<[title]>` in I18N_STORY_VIEWER_COMPLETED_CHAPTER? Ditto for languages in the next field.

This will avoid the translations having to be retranslated when the numbers change. If you do this, please also add a note in qqq.json that the `<[...]>` part represents a number and should be left as-is when translating, though its position in the translated text may change.

aks681 replied:
Done

seanlip commented on assets/i18n/en.json:
```diff
@@ -519,16 +519,41 @@
+    "I18N_SPLASH_STUDENTS_TITLE": "An Immersive Environment",
+    "I18N_SPLASH_STUDENT_DETAILS_1": "- Mira, Student, Palestine",
+    "I18N_SPLASH_STUDENT_DETAILS_2": "- Sama, Student, Palestine",
+    "I18N_SPLASH_STUDENT_DETAILS_3": "- Sara, Student, Palestine",
```
Shouldn't this change to the student from India?

aks681 replied:
Done

seanlip commented on assets/i18n/en.json:
```diff
@@ -519,16 +519,41 @@
+    "I18N_SPLASH_TEACHERS_TITLE": "Easily Share Your Knowledge",
+    "I18N_SPLASH_TESTIMONIAL_1": "“I enjoyed playing the lesson a lot, I did not feel bored and I feel like I master the negative numbers now”",
+    "I18N_SPLASH_TESTIMONIAL_2": "“I learned ratios at school but with Oppia I was introduced to many new things like the three items ratios and combining ratios”",
+    "I18N_SPLASH_TESTIMONIAL_3": "“I love Oppia so much, and I hope that more topics like geography and science are available in Arabic”",
```
This one should change too, right?

aks681 replied:
Done

seanlip commented on assets/i18n/en.json:
```diff
@@ -519,16 +519,41 @@
     "I18N_SOLICIT_ANSWER_DETAILS_FEEDBACK": "Okay, now let's go back to your answer.",
     "I18N_SOLICIT_ANSWER_DETAILS_QUESTION": "Could you explain why you picked this answer?",
+    "I18N_SPLASH_BENEFITS_ONE": "Personalized Learning",
+    "I18N_SPLASH_BENEFITS_THREE": "Easy-to-follow Lessons",
```
Capitalize "Follow"

aks681 replied:
Done

seanlip commented on assets/i18n/en.json:
```diff
@@ -519,16 +519,41 @@
     "I18N_SPLASH_SITE_FEEDBACK": "Site Feedback",
-    "I18N_SPLASH_SUBTITLE": "Oppia makes it easy to create interactive lessons that educate and engage.",
+    "I18N_SPLASH_START_CONTRIBUTING": "Start Contributing",
+    "I18N_SPLASH_STUDENTS_CONTENT": "Oppia identifies common wrong answers and provides tailored feedback, so that students get a personalized experience. Our lessons retain engagement through playful characters that lead students and diverse strategies that solidify knowledge. Check out our math lessons with proven results!",
```
Change second sentence to "Our lessons keep students engaged through playful characters, and use different strategies to solidify their knowledge." This probably needs further work in the future but at least it's less jargon-y.

aks681 replied:
Done

seanlip commented on assets/i18n/en.json:
```diff
@@ -519,16 +519,41 @@
-    "I18N_SPLASH_TITLE": "Think outside the books.",
+    "I18N_SPLASH_TITLE_LINE_1": "Oppia:",
+    "I18N_SPLASH_TITLE_LINE_2": "Learning for Everyone",
+    "I18N_SPLASH_VOLUNTEERS_CONTENT": "No matter who you are, you can find a home at Oppia. We always need more people to improve lessons by suggesting questions, submitting graphics, or translating lessons.",
```
submitting --> contributing

aks681 replied:
Done

seanlip commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -29,47 +30,74 @@ const splashConstants = require('constants.ts');
+        return WindowDimensionsService.isWindowNarrow();
+      };
+
+      // The 2 functions below are to cycle between values: 0, 1, 2 for
```
Better to generalize this by having a constant for the total number of testimonials, since we may add more in the future and you don't want to have to change that number everywhere.

aks681 replied:
Done

seanlip commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -29,47 +30,74 @@ const splashConstants = require('constants.ts');
+      };
+
+      ctrl.getTestimonialBackgroundUrl = function() {
+        if (ctrl.isWindowNarrow()) {
```
Couldn't you just use CSS media queries to achieve this? Ditto below.

aks681 replied:
You mean, move all of this to CSS? I couldn't do that since the URL itself requires calling `getStaticImageUrl()` function, which I don't think is possible via CSS.

seanlip replied:
But is it possible to hardcode the URL in the CSS?

aks681 replied:
It would be possible to hardcode, but we don't do that for images in general, so not sure if we should do this here.

seanlip replied:
Per our discussion, we do it already (e.g. for the svg diagram editor and music notes input), so I think doing it here would make sense.

seanlip commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -29,47 +30,74 @@ const splashConstants = require('constants.ts');
       ctrl.$onInit = function() {
         ctrl.userIsLoggedIn = null;
+        ctrl.testimonialId = 0;
+        ctrl.classroomUrl = (
```
Use UrlInterpolationService.

aks681 replied:
Done

seanlip commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -29,47 +30,74 @@ const splashConstants = require('constants.ts');
+
+      // The 2 functions below are to cycle between values: 0, 1, 2 for
+      // testimonialId.
+      ctrl.incrementTestimonialId = function() {
```
Throughout, maybe change "testimonial id" to "displayed testimonial id"

aks681 replied:
Done

seanlip reviewed (changes requested):
Thanks @aks681! Just minor comments.

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -29,47 +30,74 @@ const splashConstants = require('constants.ts');
+      };
+
+      ctrl.decrementTestimonialId = function() {
+        // This makes sure that decrementing from 0, returns 2 instead of -1,
+        // since we want the testimonials to cycle through.
+        ctrl.testimonialId = (((ctrl.testimonialId - 1) % 3) + 3) % 3;
```
```suggestion
        // This makes sure that decrementing from 0, returns 2 instead of -1,
        // since we want the testimonials to cycle through.
        ctrl.testimonialId = (ctrl.testimonialId - 1) % 3;
```
This should work too right?

aks681 replied:
No, I had tried this, but (0 - 1) % 3 returns -1 in the JS console.

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.html:
```diff
@@ -1,174 +1,354 @@
+<div class="oppia-splash-section-three">
+  <div class="oppia-splash-section-three-left">
+    <h3 ng-if="$ctrl.isWindowNarrow()" translate="I18N_SPLASH_BENEFITS_TITLE"></h3>
+    <img ng-src="<[$ctrl.getStaticImageUrl('/splash/matthew_fractions.png')]>" class="splash-matthew-image" alt="">
```
For non-SVG images use `<picture>` and provide WebP version of the image too. See [https://web.dev/serve-images-webp/](https://web.dev/serve-images-webp/) for reference.

aks681 replied:
Done

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -29,47 +30,74 @@ const splashConstants = require('constants.ts');
-          WindowRef.nativeWindow.location = '/creator-dashboard?mode=create';
-        }, 150);
-        return false;
+      ctrl.getMainBackgroundUrl = function() {
+        if (ctrl.isWindowNarrow()) {
+          return ctrl.getStaticImageUrl('/splash/splashMainMobile.png');
+        } else {
+          return ctrl.getStaticImageUrl('/splash/splashMainDesktop.png');
+        }
+      };
```
This should be achievable by using `<picture>` stage that can pick a version of the image depending on the size of the display.

aks681 replied:
It does work here (changed it), though it doesn't work for the other functions as we need it as part of the 'background-url' css property and it is not a normal <img> tag.

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.html:
```diff
@@ -1,174 +1,354 @@
-           translate="I18N_SPLASH_THIRD_EXPLORATION_DESCRIPTION"
-           dir="auto">
-      </div>
+<div class="oppia-splash-section-five" ng-style="{'background-image': '<[$ctrl.getTestimonialBackgroundUrl()]>'}">
```
Could this section be generalized so that instead of checking for `$ctrl.testimonialId === x` we would request the required data from some dictionary? I think this approach is used on the landing page.

aks681 replied:
Done

vojtechjelinek reviewed (changes requested):
Thanks! Few comments.

seanlip commented:
@srijanreddy98 @jameesjohn -- just a reminder, could you PTAL as codeowners? Thanks!

srijanreddy98 commented on core/templates/css/oppia.css:
```diff
@@ -3264,35 +3264,41 @@ md-card.preview-conversation-skin-supplemental-card {
 
 .oppia-splash-section-one {
-  background-color: #afd2eb;
+  background-color: #e7f6ff;
```
From line. 3266

srijanreddy98 commented on core/templates/css/oppia.css:
```diff
@@ -3264,35 +3264,41 @@ md-card.preview-conversation-skin-supplemental-card {
 /* New Splash page */
 
 .oppia-splash-section-one {
```
[Optional] From here (line 3266) to line 3465 seems to be specific for the splash page. Can we move these styles to the Splash page component?

aks681 replied:
Done

srijanreddy98 commented on core/templates/pages/splash-page/splash-page.component.html:
```diff
@@ -1,174 +1,354 @@
 
 <style>
-  splash-page .oppia-splash-rows-container {
+  splash-page .oppia-splash-section-four {
```
[Optional] Using tag name in CSS selectors doesn't work out in Angular. Can you wrap this entire page in a div of class `splash-page` and change the selector to `.splash-page .oppia-splash-section-four`. And similarly at other places.

aks681 replied:
Done

srijanreddy98 reviewed (approved):
I have two optional comments but otherwise LGTM!

jameesjohn reviewed (approved):
LGTM

aks681 committed:
Addressed comments

aks681 committed:
Addressed more comments

aks681 committed:
review comment

aks681 committed:
Merge remote-tracking branch 'upstream/develop' into splash-page

aks681 committed:
addressed review comments and minor UI changes

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.html:
```diff
@@ -1,201 +1,727 @@
+      <h3 ng-if="$ctrl.isWindowNarrow()" translate="I18N_SPLASH_BENEFITS_TITLE"></h3>
+      <picture class="splash-matthew-image">
+        <source type="image/webp" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/matthew_fractions.webp')]>">
+        <source type="image/jpeg" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/matthew_fractions.png')]>">
```
```suggestion
        <source type="image/png" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/matthew_fractions.png')]>">
```

aks681 replied:
Done

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.html:
```diff
@@ -1,201 +1,727 @@
+    <div class="oppia-splash-section-four-image-mobile" ng-if="$ctrl.isWindowNarrow()">
+      <picture class="splash-student-desk-image">
+        <source type="image/webp" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/student_desk_mobile.webp')]>">
+        <source type="image/jpeg" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/student_desk_mobile.png')]>">
```
```suggestion
        <source type="image/png" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/student_desk_mobile.png')]>">
```

aks681 replied:
Done

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -29,47 +30,106 @@ import splashConstants from 'assets/constants';
+        }, {
+          quote: $translate.instant('I18N_SPLASH_TESTIMONIAL_2'),
+          studentDetails: $translate.instant('I18N_SPLASH_STUDENT_DETAILS_2'),
+          imageUrl: '/splash/Dheeraj_3.jpeg',
```
Should we unify the naming and the format of these images?

aks681 replied:
Done, converted jpeg to pngs.

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.html:
```diff
@@ -1,201 +1,727 @@
+    <div class="oppia-splash-section-seven-right" ng-if="!$ctrl.isWindowNarrow()">
+      <picture class="splash-vol-image">
+        <source type="image/webp" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/volunteer_section.webp')]>">
+        <source type="image/jpeg" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/volunteer_section.png')]>">
```
```suggestion
        <source type="image/png" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/volunteer_section.png')]>">
```

aks681 replied:
Done

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.html:
```diff
@@ -1,201 +1,727 @@
+    <div class="oppia-splash-section-seven-image" ng-if="$ctrl.isWindowNarrow()">
+      <picture class="splash-vol-image">
+        <source type="image/webp" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/volunteer_section.webp')]>">
+        <source type="image/jpeg" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/volunteer_section.png')]>">
```
```suggestion
        <source type="image/png" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/volunteer_section.png')]>">
```

aks681 replied:
Done

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.html:
```diff
@@ -1,201 +1,727 @@
+    <div class="oppia-splash-section-six-left">
+      <picture class="splash-books-image">
+        <source type="image/webp" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/books_dsk.webp')]>">
+        <source type="image/jpeg" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/books_dsk.png')]>">
```
```suggestion
        <source type="image/png" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/books_dsk.png')]>">
```

aks681 replied:
Done

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.html:
```diff
@@ -1,201 +1,727 @@
+    <div class="oppia-splash-section-four-right" ng-if="!$ctrl.isWindowNarrow()">
+      <picture class="splash-student-desk-image">
+        <source type="image/webp" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/student_desk.webp')]>">
+        <source type="image/jpeg" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/student_desk.png')]>">
```
```suggestion
        <source type="image/png" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/student_desk.png')]>">
```

aks681 replied:
Done

vojtechjelinek commented on core/templates/pages/splash-page/splash-page.component.html:
```diff
@@ -1,201 +1,727 @@
       </div>
+    </div>
+    <div class="oppia-splash-section-one-right">
+      <picture class="oppia-splash-main-image image-desktop">
+        <source type="image/webp" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/splashMainDesktop.webp')]>">
+        <source type="image/jpeg" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/splashMainDesktop.png')]>">
+        <img ng-src="<[$ctrl.getStaticImageUrl('/splash/splashMainDesktop.png')]>" alt="">
+      </picture>
+      <picture class="oppia-splash-main-image image-mobile">
+        <source type="image/webp" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/splashMainMobile.webp')]>">
+        <source type="image/jpeg" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/splashMainMobile.png')]>">
+        <img ng-src="<[$ctrl.getStaticImageUrl('/splash/splashMainMobile.png')]>" alt="">
+      </picture>
```
This should work too and should be more efficient (in terms of not downloading unneeded images).
```suggestion
      <picture class="oppia-splash-main-image">
        <source type="image/webp" media="(min-width: 769px)" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/splashMainDesktop.webp')]>">
        <source type="image/jpeg" media="(min-width: 769px)" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/splashMainDesktop.png')]>">
        <source type="image/webp" media="(max-width: 768px)" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/splashMainMobile.webp')]>">
        <source type="image/jpeg" media="(max-width: 768px)" ng-srcset="<[$ctrl.getStaticImageUrl('/splash/splashMainMobile.png')]>">
        <img ng-src="<[$ctrl.getStaticImageUrl('/splash/splashMainMobile.png')]>" alt="">
      </picture>
```

aks681 replied:
I tried this, but it showed both images on both desktop and mobile, so reverted this back.

vojtechjelinek replied:
Oh, I edited it now, but feel free to include it in follow-up PR?

vojtechjelinek reviewed (changes requested):
Thanks! Left a few comments.

dchen97 commented:
Thanks Akshay. Some last comments:

- Is it intended for the hero photo / area to be so large - especially in desktop? The photo and area taken up seems to be disproportionately large to the text and CTA. I think this could be addressed by potentially having a second (non-primary) CTA (potentially linked to the library?) but in generally, it seems to take up way too much space because of the photo on the right. We should make that smaller and more proportional to the text size.
- We should change the anchor text that says "Learn more" to be more descriptive, as it helps users better scan the page and helps with SEO as well. The first "learn more" can be changed to "Start Learning" and the second could be "Start Teaching"
- "Immersive Environment" is too complicated and generally is not appealing to students. Should change to "Fun and effective learning"

aks681 committed:
Merge remote-tracking branch 'upstream/develop' into splash-page

seanlip commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -30,45 +30,45 @@ import splashConstants from 'assets/constants';
-        // since we want the testimonials to cycle through.
-        ctrl.testimonialId = (ctrl.testimonialId + 1) % 3;
+      // The 2 functions below are to cycle between values:
+      // 0 to testimonialCount - 1 for displayedTestimonialId.
```
Put parens around (testimonialCount - 1), otherwise this is a bit hard to parse.

aks681 replied:
Done

seanlip commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -30,45 +30,45 @@ import splashConstants from 'assets/constants';
+      // The 2 functions below are to cycle between values:
+      // 0 to testimonialCount - 1 for displayedTestimonialId.
+      ctrl.incrementDisplayedTestimonialId = function() {
+        // This makes sure that incrementing from testimonialCount - 1, returns
```
`from (testimonialCount - 1) returns ... `

aks681 replied:
Done

seanlip commented on core/templates/pages/splash-page/splash-page.component.ts:
```diff
@@ -30,45 +30,45 @@ import splashConstants from 'assets/constants';
+        // testimonialCount - 1 instead of -1, since we want the testimonials to
+        // cycle through.
+        ctrl.displayedTestimonialId = (
+          ((ctrl.displayedTestimonialId - 1) % ctrl.testimonialCount) +
```
Typically, just do:

    (ctrl.displayedTestimonialId + ctrl.testimonialCount - 1) % ctrl.testimonialCount

and then you save on one % operation.

aks681 replied:
Done

seanlip reviewed (changes requested):
Thanks @aks681, just a few more notes.  Please also see https://github.com/oppia/oppia/pull/11260#discussion_r544079174.

aks681 committed:
Merge remote-tracking branch 'upstream/develop' into splash-page

aks681 committed:
Addressed review comments

aks681 reviewed (commented):
Re @dchen97:

- I think it is supposed to be quite large. In a call with Christine, she had asked for the title to be centred, so it looks better, so did that.
- Rest 2 done

aks681 commented:
Modified Video:



https://user-images.githubusercontent.com/22347970/102695518-e95c7e00-424d-11eb-9808-15243f2aa3f9.mp4

aks681 commented:
@U8NWXD Just wanted to note that the PR used to fail lint checks (in the sense that the script errored out and didn't give an output) but after I removed .jpeg files and replaced them with .png files, this no longer happens and lint runs fine.
Not sure if the images were the reason, but just wanted to point it out.

seanlip reviewed (approved):
Thanks @aks681! LGTM. Let's please try to get this in ASAP, it's running late already.

vojtechjelinek reviewed (approved):
LGTM!

aks681 commented:
The CircleCI tests have passed in the dashboard (https://app.circleci.com/pipelines/github/aks681/oppia?branch=splash-page), but for some reason, it's not synced here.

aks681 commented:
Oh, nvm, it worked now, as I commented :)

vojtechjelinek commented:
@aks681 The splash page is now broken on the backup server.
![image](https://user-images.githubusercontent.com/10142938/102894191-06b66580-4463-11eb-8e14-4e56a548c944.png)

DubeySandeep commented:
Thanks for reporting this @vojtechjelinek, I'm removing the `PR: for current release` label as this needs more testing before we release and to avoid last minute bug fixing in the Dec release. (@aks681 let me know if it's important for this release we can discuss.)

aks681 commented:
I just checked and it seems to be working fine there..
![image](https://user-images.githubusercontent.com/22347970/102894956-e2ba4a80-4489-11eb-8226-5ea162fccddb.png)

Probably a race condition somewhere, as I did a couple of refreshes and it showed correctly all the time, I'll see if I can reproduce this and create a PR for it.

vojtechjelinek commented:
@aks681 Ctrl + F5 refresh fixed this for me
